frameworks/fmi2api/src/main/java/org/intocps/maestro/framework/fmi2/api/mabl/scoping/ScopeFmi2Api.java

```diff
@@ -219,6 +216,92 @@ protected StringVariableFmi2Api store(Supplier<String> nameProvider, String valu
                 newAIdentifierExp(name));
     }
 
+    /**
+     *
+     * @param identifyingName the name of the array
+     * @param mdArray non-jagged multidimensional array.
+     * @param <V> data type
+     * @return an ArrayVariable representing the multidimensional array
+     */
+    private <V> ArrayVariableFmi2Api<V> createMDArray(String identifyingName, V[] mdArray) {
+        List<Integer> arrayShape = new ArrayList<>();
+        PType type;
+        V[] subArr = mdArray;
+        while(subArr.getClass().getComponentType().isArray()){
+            arrayShape.add(subArr.length);
+            subArr = (V[])subArr[0];
+        }
+        arrayShape.add(subArr.length);
+
+        if (subArr instanceof Double[]) {
+            type = newARealNumericPrimitiveType();
+        } else if (subArr instanceof Integer[]) {
+            type = newAIntNumericPrimitiveType();
+        } else if (subArr instanceof Boolean[]) {
+            type = newABoleanPrimitiveType();
+        } else if (subArr instanceof String[]) {
+            type = newAStringPrimitiveType();
```

Comment:
is this all types. Isn't there also a UInt type?

Reply:
I just implemented support for the Java types found in the existing store method. However there is a 'AUIntNumericPrimitiveType'.  As i understand it newAUIntNumericPrimitiveType() should be used when the Java datatype is of type Long as Java doesn't have a datatype for unsigned integers?